envoy/network/listener.h

```diff
@@ -428,6 +443,45 @@ class UdpListener : public virtual Listener {
 
 using UdpListenerPtr = std::unique_ptr<UdpListener>;
 
+/**
+ * Internal listener callbacks.
+ */
+class InternalListenerCallbacks {
```

Comment:
I think this should be named `InternalListener`. `InternalListenerCallbacks` would be an interface passed to an `InternalListener` so that it can access other parts of the system.